src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -50,6 +52,23 @@ case class Content(root: Directory, fingerprint: NoOpFingerprint = NoOpFingerpri
 
     rec(new Path(root.name), root)
   }
+
+  @JsonIgnore
+  lazy val fileInfos: Set[FileInfo] = {
+    // Recursively find files from directory tree.
+    def rec(prefixPath: Path, directory: Directory): Seq[FileInfo] = {
+      val files = directory.files.map(f =>
+        FileInfo(new Path(prefixPath, f.name).toString, f.size, f.modifiedTime))
+      files ++ directory.subDirs.flatMap { dir =>
+        rec(new Path(prefixPath, dir.name), dir)
+      }
+    }
+    if (root.name.isEmpty) {
+      Set()
```

Comment:
This should never happen if the design and previous code is correct. `root.name` should always be `file:/` for unix, `file:/c:/` for windows or something similar. Did you face an issue where `root.name.isEmpty == true`? If that's the case, it's a bug in the code somewhere.

Could you throw an exception here instead of returning empty Seq()?

(I may have missed some other context which makes this situation appear. Please correct me if I am wrong)

Reply:
I added this because of the test failure. I'll change this to assert.
Thanks!